test/value.test.ts

```diff
@@ -62,8 +62,17 @@ it('parses binary integers', () => {
 	expect(parseValue('0b1101_0110', '', 0)).toBe(0b11010110)
 })
 
-it('rejects numbers too large', () => {
-	expect(() => parseValue('9007199254740992', '', 0)).toThrowError(TomlError)
+it('parses large integers as BigInt', () => {
+	// JavaScript's Number.MAX_SAFE_INTEGER is 9007199254740991
+	// Numbers larger than this should be parsed as BigInt
+	const largeInt = parseValue('9007199254740992', '', 0)
+	expect(typeof largeInt).toBe('bigint')
+	expect(largeInt).toBe(9007199254740992n)
```

Comment:
Testing the limit case is good! You could avoid hardcoding the max safe integer value by doing `(BigInt(Number.MAX_SAFE_INTEGER) + 1n).toString()`, but that's more of a nitpick